crypto/keyring/alt_signing_algorithms.go

```diff
@@ -0,0 +1,44 @@
+package keyring
+
+import "github.com/tendermint/tendermint/crypto"
+
+type AltSigningAlgo interface {
+	Name() SigningAlgo
+	DeriveKey() AltDeriveKeyFunc
+	PrivKeyGen() AltPrivKeyGenFunc
+}
+
+type AltDeriveKeyFunc func(mnemonic string, bip39Passphrase, hdPath string) ([]byte, error)
```

Comment:
Isn't this the same as `DeriveKeyFunc`?

Reply:
Not really, DeriveKeyFunc you pass in the end the algorythm.

	DeriveKeyFunc func(mnemonic string, bip39Passphrase, hdPath string, algo SigningAlgo) ([]byte, error)

Which for me is reiterative in this new design.

Reply:
Ohh true. Sorry for the noise.